### ISSUE TITLE
libvirt integration ideas

### ISSUE THREAD

**Issue by arighi** [arighi/virtme-ng#207]
Even if virtme-ng is not a VM manager, it'd be nice to have some kind of integration with libvirt, for example it could be useful to have a way to "reuse" the same qemu options of a target libvirt instance.

This shouldn't be too difficult to implement, in theory we could just run `virsh domxml-to-native qemu-argv --domain TARGET_VM` to get the exact qemu command line (without actually starting the instance) and re-use the same options with `vng --qemu-opts="..."`.

Ideally we should provide a proper vng option to do this, something like  `--libvirt-domain VM` to make this feature easy to use.

**Comment by matttbe** [comment]
Interesting! I don't regularly use `virsh`, but I guess it will be required to filter-out some options due to conflicts or because libvirt is maybe setting things up on the host that are not/might not be available when launched manually?

**Comment by arighi** [comment]
Right, some of the stuff to filter out are the disk/file/partition, since we still want to use the host's filesystem and not boot into the actual instance. Maybe initially we can just whitelist a few "safe" options... I'll play a bit with it.

**Comment by mhartmay** [comment]
Not sure how useful it is to integrate with libvirt, but as a starting point we might (also) integrate with `systemd-machined`. See https://www.freedesktop.org/software/systemd/man/latest/org.freedesktop.machine1 for the protocol using DBUS.

**Comment by mhartmay** [comment]
FYI, there is also `virt-qemu-run`, but that would mean that we use libvirt to start the QEMU guest (but without the need of having the daemon running). [1]

[1] https://libvirt.org/manpages/virt-qemu-run.html
